utilities/pipelines/resourceRemoval/helper/Invoke-ResourceRemoval.ps1

```diff
@@ -100,74 +101,98 @@ function Invoke-ResourceRemoval {
+            $backupItems = Get-AzRecoveryServicesBackupItem -BackupManagementType 'AzureVM' -WorkloadType 'AzureVM' -VaultId $ResourceId
             foreach ($backupItem in $backupItems) {
-                Write-Verbose ('Removing Backup item [{0}] from RSV [{1}]' -f $backupItem.Name, $resourceId) -Verbose
+                Write-Verbose ('Removing Backup item [{0}] from RSV [{1}]' -f $backupItem.Name, $ResourceId) -Verbose
 
                 if ($backupItem.DeleteState -eq 'ToBeDeleted') {
                     if ($PSCmdlet.ShouldProcess('Soft-deleted backup data removal', 'Undo')) {
-                        $null = Undo-AzRecoveryServicesBackupItemDeletion -Item $backupItem -VaultId $resourceId -Force
+                        $null = Undo-AzRecoveryServicesBackupItemDeletion -Item $backupItem -VaultId $ResourceId -Force
                     }
                 }
 
-                if ($PSCmdlet.ShouldProcess(('Backup item [{0}] from RSV [{1}]' -f $backupItem.Name, $resourceId), 'Remove')) {
-                    $null = Disable-AzRecoveryServicesBackupProtection -Item $backupItem -VaultId $resourceId -RemoveRecoveryPoints -Force
+                if ($PSCmdlet.ShouldProcess(('Backup item [{0}] from RSV [{1}]' -f $backupItem.Name, $ResourceId), 'Remove')) {
+                    $null = Disable-AzRecoveryServicesBackupProtection -Item $backupItem -VaultId $ResourceId -RemoveRecoveryPoints -Force
                 }
             }
 
             # Actual removal
             # --------------
-            $null = Remove-AzResource -ResourceId $resourceId -Force -ErrorAction 'Stop'
+            $null = Remove-AzResource -ResourceId $ResourceId -Force -ErrorAction 'Stop'
             break
         }
         'Microsoft.OperationalInsights/workspaces' {
-            $resourceGroupName = $resourceId.Split('/')[4]
-            $resourceName = Split-Path $resourceId -Leaf
+            $resourceGroupName = $ResourceId.Split('/')[4]
+            $resourceName = Split-Path $ResourceId -Leaf
             # Force delete workspace (cannot be recovered)
             if ($PSCmdlet.ShouldProcess("Log Analytics Workspace [$resourceName]", 'Remove')) {
+                Write-Verbose ('[*] Purging resource [{0}] of type [{1}]' -f $resourceName, $Type) -Verbose
                 $null = Remove-AzOperationalInsightsWorkspace -ResourceGroupName $resourceGroupName -Name $resourceName -Force -ForceDelete
             }
             break
         }
         'Microsoft.DevTestLab/labs' {
-            $resourceGroupName = $resourceId.Split('/')[4]
-            $resourceName = Split-Path $resourceId -Leaf
+            $resourceGroupName = $ResourceId.Split('/')[4]
+            $resourceName = Split-Path $ResourceId -Leaf
             if ($PSCmdlet.ShouldProcess("DevTestLab Lab [$resourceName]", 'Remove')) {
                 Get-AzResourceLock -ResourceGroupName $resourceGroupName |
```

Comment:
I guess this can be removed now?

Reply:
We could, but then we change it to use Invoke-ResourceLockRemoval, or remove it entierly as its now infront of the switch and will done on every resource.

Reply:
Removing the whole DevTestLab section in the switch statement as it is now default included in the function to remove locks up front.